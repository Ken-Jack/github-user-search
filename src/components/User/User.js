// @flow

import React from 'react';
import {connect} from 'react-fela';
import {Link} from 'react-router-dom';
import Avatar from 'components/Avatar';

type Props = {
  login: string,
  avatar_url: string,
  styles: any,
};

function User({login, avatar_url, styles}: Props) {
  return (
    <Link
      to={login}
      className={styles.root}
    >
      <div className={styles.avatar}>
        <Avatar
          url={avatar_url}
          name={login}
          width={90}
        />
      </div>
      <p className={styles.username}>{login}</p>
    </Link>
  );
}

const styles = {
  root: () => ({
    color: 'inherit',
    display: 'flex',
  }),

  avatar: () => ({
    marginRight: '15px',
  }),

  username: () => ({
    fontSize: '20px',
  }),
};

export default connect(styles)(User);
