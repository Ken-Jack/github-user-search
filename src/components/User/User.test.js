import React from 'react';
import {
  StaticRouter as Router,
} from 'react-router-dom';
import felaSnapshot from 'test-util/fela-snapshot';
import User from './User';

describe('Component: User', () => {

  it('should render', () => {
    const component = felaSnapshot(
      <Router context={{}}>
        <User
          login="simon"
          avatar_url="url"
        />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });

});
