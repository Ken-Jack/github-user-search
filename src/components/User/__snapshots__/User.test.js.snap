// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: User should render 1`] = `
Object {
  "component": <a
    className="a b"
    href="/simon"
    onClick={[Function]}
  >
    <div
      className="c"
    >
      <div
        className="f"
        style={
          Object {
            "minHeight": 60,
          }
        }
      >
        <img
          alt="simon's avatar"
          className="e k l m"
          onLoad={[Function]}
          src="url&s=180"
          theme={Object {}}
          width={90}
        />
        <svg />
      </div>
    </div>
    <p
      className="d"
    >
      simon
    </p>
  </a>,
  "styles": "
    .a{color:inherit}
    .b{display:flex}
    .c{margin-right:15px}
    .d{font-size:20px}
    .e{border-radius:8px}
    .f{position:relative}
    .g{left:50%}
    .h{top:50%}
    .i{position:absolute}
    .j{transform:translate(-50%, -50%)}
    .k{display:block}
    .l{min-width:100%}
    .m{height:auto}",
}
`;
